storm-core/src/jvm/backtype/storm/messaging/netty/KerberosSaslClientHandler.java

```diff
@@ -46,56 +44,48 @@ public KerberosSaslClientHandler(ISaslClient client, Map storm_conf, String jaas
     }
 
     @Override
-    public void channelConnected(ChannelHandlerContext ctx,
-                                 ChannelStateEvent event) {
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
         // register the newly established channel
-        Channel channel = ctx.getChannel();
-        client.channelConnected(channel);
+        Channel channel = ctx.channel();
+        client.channelReady();
 
         LOG.info("Connection established from {} to {}",
-                 channel.getLocalAddress(), channel.getRemoteAddress());
+                 channel.localAddress(), channel.localAddress());
```

Comment:
I don't think channel.getRemoteAddress and channel.localAddress return the same thing.